velox/vector/arrow/Bridge.cpp

```diff
@@ -1866,6 +1866,24 @@ VectorPtr createShortDecimalVector(
       pool, type, nulls, length, values, nullCount);
 }
 
+// Arrow uses two uint64_t values to represent a 128-bit decimal value. The
+// memory allocated by Arrow might not be 16-byte aligned, so we need to copy
+// the values to a new buffer to ensure 16-byte alignment.
+VectorPtr createLongDecimalVector(
+    memory::MemoryPool* pool,
+    const TypePtr& type,
+    BufferPtr nulls,
+    const int128_t* input,
+    size_t length,
+    int64_t nullCount) {
+  auto values = AlignedBuffer::allocate<int128_t>(length, pool);
```

Comment:
Can you check if the memory address is 16-byte aligned? Sometimes it is 16-byte aligned, this is not a common case, I check it by https://github.com/apache/incubator-gluten/blob/main/cpp/velox/shuffle/VeloxShuffleReader.cc#L113 in gluten.

I meet the case because Arrow function SliceBuffer returns the buffer that is not aligned, https://github.com/apache/arrow/blob/release-15.0.0-rc1/cpp/src/arrow/buffer.h#L111

Reply:
Sorry for the late reply, I mean check the `input` alignment, we may not need to copy if aligned.